__tests__/app.test.js

```diff
@@ -65,3 +65,90 @@ describe("GET /api/articles/:article_id", () => {
+    const response = await request(app).get("/api/articles/1/comments");
+    expect(response.status).toBe(200);
+    const comments = response.body.comments;
+    expect(comments.length).toBe(11);
+    comments.map((comment) => {
+      expect(comment).toHaveProperty("comment_id");
+      expect(comment).toHaveProperty("body");
+      expect(comment).toHaveProperty("article_id");
+      expect(comment).toHaveProperty("author");
+      expect(comment).toHaveProperty("votes");
+      expect(comment).toHaveProperty("created_at");
+    });
+    expect(comments).toBeSortedBy("created_at");
+  });
+  it("404: responds the appropriate message when article_id is valid but not in the database", async () => {
+    const response = await request(app).get("/api/articles/999/comments");
+    expect(response.status).toBe(404);
+    expect(response.body.msg).toBe("Not Found");
+  });
+  it("400: responds the appropriate message when article_id is not valid", async () => {
+    const response = await request(app).get("/api/articles/invalid/comments");
+    expect(response.status).toBe(400);
+    expect(response.body.msg).toBe("Bad Request");
+  });
+});
+
+describe("POST /api/articles/:article_id/comments", () => {
+  it("201: responds with the posted comment and posts a new comment", async () => {
+    const newComment = { username: "butter_bridge", body: "testing the body" };
+    const response = await request(app)
+      .post("/api/articles/10/comments")
+      .send(newComment);
+    expect(response.status).toBe(201);
+    const comment = response.body.comment;
+    expect(comment.author).toBe(newComment.username);
+    expect(comment.body).toBe(newComment.body);
+    expect(comment.article_id).toBe(10);
+  });
+  it("404: responds with appropriate message when article_id is not found", async () => {
+    const newComment = { username: "butter_bridge", body: "testing the body" };
+    const response = await request(app)
+      .post("/api/articles/999/comments")
+      .send(newComment);
+    expect(response.status).toBe(404);
+    expect(response.body.msg).toBe("article not found");
+  });
+  it("400: responds with appropriate message when posted comment does not contain required properties", async () => {
+    const newComment = { body: "testing the body" };
+    const response = await request(app)
+      .post("/api/articles/10/comments")
+      .send(newComment);
+    expect(response.status).toBe(400);
+    expect(response.body.msg).toBe("Bad Request. Missing properties.");
+  });
+  it("400: responds with appropriate message when posted comment contains invalid property values", async () => {
+    const newComment = { username: 8777, body: 67 };
+    const response = await request(app)
+      .post("/api/articles/10/comments")
+      .send(newComment);
+    expect(response.status).toBe(400);
+    expect(response.body.msg).toBe("Bad Request");
+  });
+});
```

Comment:
Nice testing here, we could also consider:
- 400 for invalid id
- 404 for username doesn't exist

Reply:
Thanks for the feedback Richard. Apologies for the mix-up with the previous tasks. Would it be ok if i apply the feedback on 'main' and then create the new tickets?

Reply:
Once you've received the feedback that's up to you how to implement it. It's always good to use branching tools to keep our commits and code organised though :)